src/backend/commands/tablecmds.c

```diff
@@ -6512,6 +6512,8 @@ ATRewriteTable(AlteredTableInfo *tab, Oid OIDNewHeap, LOCKMODE lockmode)
 		oldslot = MakeSingleTupleTableSlot(oldTupDesc);
 		newslot = MakeSingleTupleTableSlot(newTupDesc);
 
+		Insist(newTupDesc && oldTupDesc);
```

Comment:
The newTupDesc and oldTupDesc pointers are dereferenced in the MakeSingleTupleTableSlot function above.

Reply:
Done